docs/src/main/asciidoc/panache-rest.adoc

```diff
@@ -0,0 +1,148 @@
+    </dependency>
+</dependencies>
+----
+
+* Implement the Panache entities and/or repositories as explained in the link:hibernate-orm-panache[Panache with Hibernate ORM guide].
+* Define the interfaces for generation as explained in the sections below.
+
+== Generating CRUD resources
+
+Panache REST generates CRUD resources based on the interfaces available in your application.
+For each entity and repository that you want to generate, provide an interface that extends either `PanacheEntityCrudResource` or `PanacheRepositoryCrudResource` interface.
+_Do not implement these interfaces and don't provide custom methods because they won't be implemented._
+You can, however, override the methods from the extended interface in order to customize them (see the section at the end).
+
+The resources that will be generated will be functionally equivalent for both entities and repositories.
+Only the data will accessed using either one or the other.
+The skeleton of the generated JAX-RS resource will look like this:
+
+[source,java]
+----
+@Path("/entities") // Base bath is a pluralised name of an entity
+public class EntityCrudResourceImpl implements EntityCrudResource { // The actual class name is going to be unique
+    @GET
+    @Produces("application/json")
+    @Path("{id}")
+    public Entity get(@PathParam("id") ID id){
+        // ...
+    }
+
+    @GET
+    @Produces("application/json")
+    public List<Entity> list(){
+        // ...
+    }
+
+    @Transactional
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    public Response add(Entity entity) {
+        // ..
+    }
+
+    @Transactional
+    @PUT
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("{id}")
+    public Response update(@PathParam("id") ID id, Entity entity) {
+        // ..
+    }
+
+    @Transactional
+    @DELETE
+    @Path("{id}")
+    public void delete(@PathParam("id") ID id) {
+        // ..
+    }
+}
+----
+
+=== PanacheEntityCrudResource
+
+If your application has an entity (e.g. `Person`) that extends either `PanacheEntity` or `PanacheEntityBase` class, you could tell Panache REST to generate its CRUD resource with the following interface:
+
+[source,java]
+----
+public interface PersonCrudResource extends PanacheEntityCrudResource<Person, Long> {
+}
+----
+
+=== PanacheRepositoryCrudResource
+
+If your application has a simple entity (e.g. `Person`) and a repository (e.g. `PersonRepository`) that implements either `PanacheRepository` or `PanacheRepositoryBase` interface, you could tell Panache REST to generate its CRUD resource with the following interface:
+
+[source,java]
+----
+public interface PersonCrudResource extends PanacheRepositoryCrudResource<PersonRepository, Person, Long> {
+}
+----
+
+== Resource customisation
+
+Panache REST provides a `@PanacheRestResource` annotation that can be used to customize certain features of the resource.
```

Comment:
It's out of scope for this PR but I'm not happy with a name of this annotation (`@PanacheRestResource`). It's very similar to the interface name (`PanacheCrud..Resource`) and might be confusing. Any suggestions on the name change @geoand?

Reply:
I dunno.... I'm probably the worst developer when it comes to naming...